habitat/core/dataset.py

```diff
@@ -101,3 +105,140 @@ def default(self, object):
+        max_episodes_per_split = min(
+            max_episodes_per_split, (len(self.episodes) // num_splits)
+        )
+        if num_splits * max_episodes_per_split == len(self.episodes):
+            # Special case for exact size match
+            stride = len(self.episodes) // num_splits
+            for ii, split in enumerate(
+                range(0, len(self.episodes), stride)[:num_splits]
+            ):
+                new_dataset = copy.copy(self)  # Creates a shallow copy
+                new_dataset.episodes = new_dataset.episodes[
+                    split : split + stride
+                ].copy()
+                if collate_scene_ids:
+                    new_dataset.episodes.sort(key=lambda ep: ep.scene_id)
+                else:
+                    random.shuffle(new_dataset.episodes)
+                new_datasets.append(new_dataset)
+            new_episodes = self.episodes
+        else:
+            rand_items = np.random.choice(
+                len(self.episodes),
+                num_splits * max_episodes_per_split,
+                replace=False,
+            )
+            if collate_scene_ids:
+                scene_ids = {}
+                for rand_ind in rand_items:
+                    scene = self.episodes[rand_ind].scene_id
+                    if scene not in scene_ids:
+                        scene_ids[scene] = []
+                    scene_ids[scene].append(rand_ind)
+                rand_items = []
+                list(map(rand_items.extend, scene_ids.values()))
+            ep_ind = 0
+            new_episodes = []
+            for nn in range(num_splits):
+                new_dataset = copy.copy(self)  # Creates a shallow copy
+                new_dataset.episodes = []
+                new_datasets.append(new_dataset)
+                for ii in range(max_episodes_per_split):
+                    new_dataset.episodes.append(
+                        self.episodes[rand_items[ep_ind]]
+                    )
+                    ep_ind += 1
+                new_episodes.extend(new_dataset.episodes)
+        if remove_unused_episodes:
+            self.episodes = new_episodes
+        return new_datasets
+
+    def get_uneven_splits(self, num_splits):
+        """
+        Returns a list of new datasets, each with a subset of the original
+        episodes. The last dataset may have fewer episodes than the others.
+        This is especially useful for splitting over validation/test datasets
+        in order to make sure that all episodes are copied but none are
+        duplicated.
+        Args:
+            num_splits: The number of splits to create.
+        Returns:
+            A list of new datasets, each with their own subset of episodes.
+        """
+        assert (
+            len(self.episodes) >= num_splits
+        ), "Not enough episodes to create this many splits."
+        num_splits = min(num_splits, len(self.episodes))
```

Comment:
Based on assert above `num_splits == num_splits`.

Reply:
Yep. I was indecisive about which behavior I wanted. I can take the min thing out.